flash_attn/models/llama.py

```diff
@@ -308,7 +309,28 @@ def config_from_meta_checkpoint(
         num_attention_heads=params["n_heads"],
         num_hidden_layers=params["n_layers"],
         rms_norm_eps=params["norm_eps"],
+        num_key_value_heads=params.get("n_kv_heads", None),
     )
+    multiple_of = params.get("multiple_of", 1)
+    ffn_dim_multiplier = params.get("ffn_dim_multiplier", None)
+
+    intermediate_size = 4 * config.hidden_size
+    intermediate_size = int(2 * intermediate_size / 3)
+    # custom dim factor multiplier
+    if ffn_dim_multiplier is not None:
+        intermediate_size = int(ffn_dim_multiplier * intermediate_size)
+    intermediate_size = multiple_of * ((intermediate_size + multiple_of - 1) // multiple_of)
+    config.intermediate_size = intermediate_size
+    if "rope_theta" in params:
+        config.rotary_emb_base = params["rope_theta"]
+    config.vocab_size = 32000
+    # CodeLLaMa
+    if "code" in model_name.lower():
+        # some have vocab_size 32000, some 32016
```

Comment:
Do you mean the vocab size varies by model size?

Reply:
Code llama has a few more tokens, partly due to the [fill-in-the-middle](https://arxiv.org/abs/2207.14255) fine-tuning. In their paper:

> "We extend Llama 2’s tokenizer with four special tokens that mark the beginning of the prefix, the middle part or the suffix, and the end of the infilling span."

In addition, there's adjustments to the sentencepiece formulation.

Reply:
For clarity, I'm just saying there's no issues with this code in terms of being consistent with the recent Hugging Face integration.